src/units.js

```diff
@@ -0,0 +1,99 @@
+/* eslint no-console:0 */
+
+/**
+ * This file does conversion between units.  In particular, it provides
+ * calculateSize to convert other units into ems.
+ */
+
+import ParseError from "./ParseError";
+
+// This table gives the number of TeX pts in one of each *absolute* TeX unit.
+// Thus, multiplying a length by this number converts the length from units
+// into pts.  Dividing the result by ptPerEm gives the number of ems
+// *assuming* a font size of ptPerEm (normal size, normal style).
+const ptPerUnit = {
+    // https://en.wikibooks.org/wiki/LaTeX/Lengths and
+    // https://tex.stackexchange.com/a/8263
+    "pt": 1,            // TeX point
+    "mm": 7227 / 2540,  // millimeter
+    "cm": 7227 / 254,   // centimeter
+    "in": 72.27,        // inch
+    "bp": 803 / 800,    // big (PostScript) points
+    "pc": 12,           // pica
+    "dd": 1238 / 1157,  // didot
+    "cc": 14856 / 1157, // cicero (12 didot)
+    "nd": 685 / 642,    // new didot
+    "nc": 1370 / 107,   // new cîcero (12 new didot)
+    "sp": 1 / 65536,    // scaled point (TeX's internal smallest unit)
+    // https://tex.stackexchange.com/a/41371
+    "px": 803 / 800,    // \pdfpxdimen defaults to 1 bp in pdfTeX and LuaTeX
```

Comment:
`px` is built into `pdflatex`.  Maybe I'm biased because that's the only TeX format I use, but that seems pretty standard/default to me.  I'd be OK with **omitting** `px` (I frankly didn't realize it was valid in `pdflatex` until working on this PR), if we think it's too confusing -- but I don't think it's a good idea for some TeX units to map to CSS units when some TeX units to map to TeX units (in some sense).